perception/map_based_prediction/src/map_based_prediction_node.cpp

```diff
@@ -1141,7 +1141,7 @@ bool MapBasedPredictionNode::doesPathCrossFence(
   const PredictedPath & predicted_path, const lanelet::ConstLineString3d & fence_line)
 {
   // check whether the predicted path cross with fence
-  for (size_t i = 0; i < predicted_path.path.size(); ++i) {
+  for (size_t i = 0; i < predicted_path.path.size() - 1; ++i) {
```

Comment:
[Q]
The previous code seems to cause invalid memory access. Did you encountered that error?

Reply:
[A]
I just found that it accesses outside of the vector.